### ISSUE TITLE
Federation gateway deployed in lambda returns error in @apollo/gateway 0.46.0 but works in 0.16.0

### ISSUE THREAD

**Issue by alina-fin** [apollographql/federation#1523]
I am creating a Apollo Federation instance deployed in AWS Lambda using Serverless. There are 2 subgraphs deployed in serverless lambda (products, reviews) and 1 gateway deployed in serverless lambda. Using @apollo/gateway v0.16.0 and apollo-server-lambda 2.13.1 (we'll call this **deprecated gateway**), I can successfully query for subgraph data through the gateway. However, using @apollo/gateway v0.47.0 and apollo-server-lambda 3.6.3 (we'll call this **new gateway**) I receive the following error:

```
{
    "errors": [
        {
            "message": "Cannot read property 'QUERY' of undefined",
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR",
                "exception": {
                    "stacktrace": [
                        "TypeError: Cannot read property 'QUERY' of undefined",
                        "    at QueryPlanningContext.getVariableUsages (/var/task/node_modules/@apollo/query-planner/dist/QueryPlanningContext.js:46:115)",
                        "    at executionNodeForGroup (/var/task/node_modules/@apollo/query-planner/dist/buildQueryPlan.js:162:36)",
                        "    at /var/task/node_modules/@apollo/query-planner/dist/buildQueryPlan.js:58:39",
                        "    at Array.map (<anonymous>)",
                        "    at buildQueryPlan (/var/task/node_modules/@apollo/query-planner/dist/buildQueryPlan.js:58:26)",
                        "    at QueryPlanner.buildQueryPlan (/var/task/node_modules/@apollo/query-planner/dist/index.js:28:52)",
                        "    at /var/task/node_modules/@apollo/gateway/dist/index.js:95:58",
                        "    at NoopContextManager.with (/var/task/node_modules/@opentelemetry/api/build/src/context/NoopContextManager.js:36:24)",
                        "    at ContextAPI.with (/var/task/node_modules/@opentelemetry/api/build/src/api/context.js:71:54)",
                        "    at NoopTracer.startActiveSpan (/var/task/node_modules/@opentelemetry/api/build/src/trace/NoopTracer.js:67:28)"
                    ]
                }
            }
        }
    ]
}
```

If I run the **new gateway** on a local server, I am able to query the same subgraphs successfully. While I understand running the gateway in serverless lambda is not recommended, the issue is that it works successfully with the deprecated versions and no longer works with the new versions. 

In summary:

| apollo/gateway version | apollo-server-lambda version | gateway location | query status |
| ----------- | ----------- | ----------- | ----------- |
| 0.16.0 |  2.13.1 | AWS Lambda | successful |
| 0.47.0  | 3.6.3  | AWS Lambda | error |
| 0.47.0 | 3.6.3 | Local | successful | 

Repositories:
Products Subgraph - https://github.com/alina-fin/productssubgraph
Reviews Subgraph - https://github.com/alina-fin/reviewssubgraph
Deprecated Gateway - https://github.com/alina-fin/deprecatedgateway
New Gateway - https://github.com/alina-fin/newgateway

Sample Query:
```
{
  "query": "query testQuery { allReviews {id rating product { upc name price }} }"
}
```

**Comment by trevor-scheer** [comment]
Hey there, thanks for the detailed explanation of your issue. We shipped a couple versions of the gateway which claimed support for `graphql@15.x` but ran into runtime errors (such as what you're seeing). I'll go ahead and deprecate those on npm now. You should be able to upgrade to the most recent gateway (`0.48.1`) without any issue, or you can keep using the same gateway but upgrade to `graphql@16.x`. Sorry for the trouble! I'm going to close this but if it happens to be the case that this doesn't solve your problem I'm happy to reopen it.

**Comment by alina-fin** [comment]
@trevor-scheer That did the trick, thank you!
